### PR TITLE
fix(tedge): `tedge config get` returns 1 when a key is not set

### PR DESCRIPTION
**Pull request by rina23q**
## Proposed changes
Return `1` if `tedge config get` is called for unset key.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
#2990 

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by Bravo555 — APPROVED** [review]

File: crates/core/tedge/src/cli/config/commands/get.rs [hunk 1]
```diff
@@ -19,6 +19,7 @@ impl Command for GetConfigCommand {
             }
             Err(tedge_config::ReadError::ConfigNotSet { .. }) => {
                 eprintln!("The provided config key: '{}' is not set", self.key);
+                std::process::exit(1)
             }
             Err(tedge_config::ReadError::ReadOnlyNotFound { message, key }) => {
                 eprintln!("The provided config key: '{key}' is not configured: {message}",);
```
Comment (didier-wenzek) [inline crates/core/tedge/src/cli/config/commands/get.rs:25]:
```suggestion
                eprintln!("The provided config key: '{key}' is not configured: {message}",);
                std::process::exit(1)
```

Even better: simply remove the case for `Err(tedge_config::ReadError::ReadOnlyNotFound {})` and let it handled by the catch all case .i.e. `Err(err) => return Err(err.into())`.

Comment (rina23q) [reply crates/core/tedge/src/cli/config/commands/get.rs:25]: The default of `clap` returns `2` for the error case. Your suggestion will change the exit code

Comment (didier-wenzek) [reply crates/core/tedge/src/cli/config/commands/get.rs:25]: I'm speaking about the second case `ReadOnlyNotFound`, not the one you fixed `ConfigNotSet`.

`ReadOnlyNotFound` must exit 1 or 2 but not 0 as today.

```
$ tedge config unset device.cert_path
$ tedge config get device.id                                                   
The provided config key: 'device.id' is not configured: The device id is read from the device certificate.
To set 'device.id' to some <id>, you can use `tedge cert create --device-id <id>`.
$ echo $?         
0
```

Comment (rina23q) [reply crates/core/tedge/src/cli/config/commands/get.rs:25]: This `execute()` is called here. 

https://github.com/thin-edge/thin-edge.io/blob/e77a5dfb64e7dd9da45b7ba2429af2d56101ba52/crates/core/tedge/src/main.rs#L65-L66

> Even better: simply remove the case for Err(tedge_config::ReadError::ReadOnlyNotFound {}) and let it handled by the catch all case .i.e. Err(err) => return Err(err.into()).

I prefer your original code suggestion, just to add `std::process::exit(1)` to `ReadOnlyNotFound` case. The reason is, if it returns `Err`, `.with_context()` will be called, that adds extra stderr output compared to now. I feel it's unnecessary change to user.

Comment (rina23q) [reply crates/core/tedge/src/cli/config/commands/get.rs:25]: Done in 0458c17271a50a6b6efb9dd9167dc5a68eeb4083

**Review by didier-wenzek — APPROVED** [review]
Approved

**Pull request merged**